kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/OperatorMainTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import io.fabric8.kubernetes.client.server.mock.KubernetesMockServer;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.search.MeterNotFoundException;
+import io.micrometer.prometheusmetrics.PrometheusMeterRegistry;
+
+import java.util.concurrent.TimeUnit;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaClusterRef;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@EnableKubernetesMockClient(crud = true)
+class OperatorMainTest {
+
+    KubernetesClient kubeClient;
+    KubernetesMockServer mockServer;
+
+    private OperatorMain operatorMain;
+
+    @BeforeEach
+    void setUp() {
+        expectApiResources();
+        operatorMain = new OperatorMain(kubeClient);
+    }
+
+    @Test
+    void shouldRegisterPrometheusMeterRegistry() {
+        // Given
+
+        // When
+        operatorMain.start();
```

Comment:
should we add a redundant operatorMain.stop() to an `@AfterEach` since this is kicking off a bunch of josdk threads

Reply:
Ohh good point about the threads yes.

Reply:
also when run on repeat in IDEA the second `shouldRemoveMetricsOnStop` fails. Odd

Reply:
think when it's run in isolation it isn't testing much because the operatorMain is never started, metrics aren't installed

Reply:
> also when run on repeat in IDEA the second `shouldRemoveMetricsOnStop` fails. Odd

Oops thats dead code, we no longer remove metrics on stop. I'd miss understood the role of prometheus registry (at least in this usage) the metrics should be on the global registry.